<!DOCTYPE html> <html lang="en"> <head> <meta http-equiv="Content-Type" content="text/html; charset=UTF-8"> <meta name="google-site-verification" content="googlede2ce4a4cee74360"> <meta http-equiv="Permissions-Policy" content="interest-cohort=()"> <meta charset="utf-8"> <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"> <meta http-equiv="X-UA-Compatible" content="IE=edge"> <title> Duc Q. Nguyen </title> <meta name="author" content="Duc Q. Nguyen"> <meta name="description" content="PhD Student at National University of Singapore. Research interests include generative models, graph representation learning, and probabilistic machine learning. "> <meta name="keywords" content="duc nguyen, machine learning, deep learning, graph neural networks, llm, nlp, nus, phd student"> <link rel="stylesheet" href="/assets/css/bootstrap.min.css?a4b3f509e79c54a512b890d73235ef04"> <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/css/mdb.min.css" integrity="sha256-jpjYvU3G3N6nrrBwXJoVEYI/0zw8htfFnhT9ljN3JJw=" crossorigin="anonymous"> <link defer rel="stylesheet" href="/assets/css/academicons.min.css?f0b7046b84e425c55f3463ac249818f5"> <link defer rel="stylesheet" href="/assets/css/scholar-icons.css?62b2ac103a88034e6882a5be5f3e2772"> <link defer rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Roboto:300,400,500,700|Roboto+Slab:100,300,400,500,700|Material+Icons&amp;display=swap"> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-github.css?591dab5a4e56573bf4ef7fd332894c99" media="" id="highlight_theme_light"> <link rel="shortcut icon" href="/assets/img/favicon.png?e7ddcaa49c1995c01d5cba46e689d73c"> <link rel="stylesheet" href="/assets/css/main.css?d41d8cd98f00b204e9800998ecf8427e"> <link rel="canonical" href="https://martinakaduc.github.io/"> <script src="/assets/js/theme.js?a81d82887dd692e91686b43de4542f18"></script> <link defer rel="stylesheet" href="/assets/css/jekyll-pygments-themes-native.css?5847e5ed4a4568527aa6cfab446049ca" media="none" id="highlight_theme_dark"> <script>
    initTheme();
  </script> </head> <body class="fixed-top-nav "> <header> <nav id="navbar" class="navbar navbar-light navbar-expand-sm fixed-top" role="navigation"> <div class="container"> <button class="navbar-toggler collapsed ml-auto" type="button" data-toggle="collapse" data-target="#navbarNav" aria-controls="navbarNav" aria-expanded="false" aria-label="Toggle navigation"> <span class="sr-only">Toggle navigation</span> <span class="icon-bar top-bar"></span> <span class="icon-bar middle-bar"></span> <span class="icon-bar bottom-bar"></span> </button> <div class="collapse navbar-collapse text-right" id="navbarNav"> <ul class="navbar-nav ml-auto flex-nowrap"> <li class="nav-item active"> <a class="nav-link" href="/">About <span class="sr-only">(current)</span> </a> </li> <li class="nav-item "> <a class="nav-link" href="/news/">News </a> </li> <li class="nav-item "> <a class="nav-link" href="/blog/">Blog </a> </li> <li class="nav-item "> <a class="nav-link" href="/projects/">Projects </a> </li> <li class="nav-item "> <a class="nav-link" href="/talks/">Talks </a> </li> <li class="nav-item "> <a class="nav-link" href="/publications/">Publications </a> </li> <li class="nav-item "> <a class="nav-link" href="/cv/">CV </a> </li> <li class="nav-item"> <button id="search-toggle" title="Search" onclick="openSearchModal()"> <span class="nav-link">ctrl k <i class="ti ti-search"></i></span> </button> </li> <li class="toggle-container"> <button id="light-toggle" title="Change theme"> <i class="ti ti-sun-moon" id="light-toggle-system"></i> <i class="ti ti-moon-filled" id="light-toggle-dark"></i> <i class="ti ti-sun-filled" id="light-toggle-light"></i> </button> </li> </ul> </div> </div> </nav> <progress id="progress" value="0"> <div class="progress-container"> <span class="progress-bar"></span> </div> </progress> </header> <div class="container mt-5" role="main"> <div class="post"> <header class="post-header"> <h1 class="post-title"> <span class="font-weight-bold">Duc</span> Q. Nguyen </h1> <p class="desc">PhD Student at <a href="https://www.nus.edu.sg/" rel="external nofollow noopener" target="_blank">National University of Singapore</a></p> </header> <article> <div class="profile float-right"> <figure> <picture> <source class="responsive-img-srcset" srcset="/assets/img/prof_pic-480.webp 480w,/assets/img/prof_pic-800.webp 800w,/assets/img/prof_pic-1400.webp 1400w," type="image/webp" sizes="(min-width: 930px) 270.0px, (min-width: 576px) 30vw, 95vw"> <img src="/assets/img/prof_pic.jpg?0100a1d6548a11af988f50c3a9b566df" class="img-fluid z-depth-1 rounded" width="100%" height="auto" alt="prof_pic.jpg" loading="eager" onerror="this.onerror=null; $('.responsive-img-srcset').remove();"> </source></picture> </figure> <div class="more-info"> <p>Singapore</p> <p><small>nqduc [at] u [dot] nus [dot] edu</small></p> </div> </div> <div class="clearfix"> <p>I am <strong>Duc Q. Nguyen</strong>, also known as <strong>Martin Nguyen</strong>, a first-year PhD student supervised by Professor <a href="https://ztangent.github.io/" rel="external nofollow noopener" target="_blank">Tan Zhi Xuan</a> at <a href="https://www.nus.edu.sg/" rel="external nofollow noopener" target="_blank">National University of Singapore</a>. My research is driven by a primary goal ‚Äì to amplify human potential through artificial intelligence. My current work focuses on <strong>human-AI alignment</strong>, <strong>generative models</strong>, and <strong>probabilistic methods</strong>. I aim to develop methods that facilitate effective human-AI collaboration to enhance capabilities in critical domains such as <strong>healthcare</strong> and <strong>education</strong>.</p> <p>Previously, I obtained my Bachelor‚Äôs and Master‚Äôs degrees from <a href="https://hcmut.edu.vn/" rel="external nofollow noopener" target="_blank">Ho Chi Minh City University of Technology (HCMUT)</a> - <a href="https://vnuhcm.edu.vn/" rel="external nofollow noopener" target="_blank">VNU-HCM</a>. During that time, I was mentored by <a href="https://cs.stanford.edu/~sttruong/" rel="external nofollow noopener" target="_blank">Sang T. Truong</a> and supervised by Professor <a href="https://sanmi.cs.illinois.edu/" rel="external nofollow noopener" target="_blank">Sanmi Koyejo</a> and Professor <a href="https://cse.hcmut.edu.vn/qttho/doku.php" rel="external nofollow noopener" target="_blank">Tho Quan</a>, working on diverse projects including drug discovery, subgraph matching, and large language models for Vietnamese.</p> </div> <h2> <a href="/news/" style="color: inherit">News</a> </h2> <div class="news"> <div class="table-responsive" style="max-height: 60vw"> <table class="table table-sm table-borderless"> <tr> <th scope="row" style="width: 20%">Oct 21, 2025</th> <td> üìú Granted patent VN.1-0053619 for ‚ÄúSystem for Real-Time Translation of Sign Language to Text and Speech‚Äù <a href="/assets/pdf/patents/VN.1-0053619.pdf">[Patent]</a>. </td> </tr> <tr> <th scope="row" style="width: 20%">Aug 03, 2025</th> <td> üéì Began doctoral studies at National University of Singapore (NUS). </td> </tr> <tr> <th scope="row" style="width: 20%">Apr 25, 2025</th> <td> üéì Graduated with Master of Engineering in Computer Science from Ho Chi Minh City University of Technology (HCMUT), Vietnam National University. </td> </tr> <tr> <th scope="row" style="width: 20%">Jan 15, 2025</th> <td> ‚úÖ Successfully defended Master‚Äôs thesis: ‚ÄúOptimizing Subgraph Isomorphism Prediction Models with Application Orientation to the Drug Design Process‚Äù. </td> </tr> <tr> <th scope="row" style="width: 20%">Sep 11, 2024</th> <td> üìù Published paper on xNeuSM in <strong>IEEE Access</strong> <a href="https://ieeexplore.ieee.org/document/10677500" rel="external nofollow noopener" target="_blank">[Paper]</a>. </td> </tr> </table> </div> </div> <h2> <a href="/blog/" style="color: inherit">Latest Posts</a> </h2> <div class="news"> <div class="table-responsive" style="max-height: 60vw"> <table class="table table-sm table-borderless"> <tr> <th scope="row" style="width: 20%">Mar 27, 2025</th> <td> <a class="news-title" href="/blog/2025/phd-application/">The Prolonged Uncertainty of the PhD Application Period</a> </td> </tr> <tr> <th scope="row" style="width: 20%">Feb 14, 2023</th> <td> <a class="news-title" href="/blog/2023/valentine-day/">The Valentine's Day Story</a> </td> </tr> <tr> <th scope="row" style="width: 20%">Apr 02, 2021</th> <td> <a class="news-title" href="/blog/2021/self-learning/">Self-learning</a> </td> </tr> </table> </div> </div> <h2> <a href="/publications/" style="color: inherit">Selected Publications</a> </h2> <div class="publications"> <ol class="bibliography"> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100" style="background-color:black"> <a href="https://www.nature.com/srep/" rel="external nofollow noopener" target="_blank">Sci. Rep.</a> </abbr> </div> <div id="nguyen_becaked_2022" class="col-sm-8"> <div class="title">BeCaked: An Explainable Artificial Intelligence Model for COVID-19 Forecasting</div> <div class="author"> <em>Duc Q. Nguyen</em>, Nghia Q. Vo, Thinh T. Nguyen, and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Khuong Nguyen-An, Quang H. Nguyen, Dang N. Tran, Tho T. Quan' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>Scientific Reports</em>, May 2022 </div> <div class="periodical"> Publisher: Nature Publishing Group </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1038/s41598-022-11693-9" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="/assets/pdf/papers/Nguyen%20et%20al.%20-%202022%20-%20BeCaked%20An%20Explainable%20Artificial%20Intelligence%20Model%20for%20COVID-19%20Forecasting.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p>From the end of 2019, one of the most serious and largest spread pandemics occurred in Wuhan (China) named Coronavirus (COVID-19). As reported by the World Health Organization, there are currently more than 100 million infectious cases with an average mortality rate of about five percent all over the world. To avoid serious consequences on people‚Äôs lives and the economy, policies and actions need to be suitably made in time. To do that, the authorities need to know the future trend in the development process of this pandemic. This is the reason why forecasting models play an important role in controlling the pandemic situation. However, the behavior of this pandemic is extremely complicated and difficult to be analyzed, so that an effective model is not only considered on accurate forecasting results but also the explainable capability for human experts to take action pro-actively. With the recent advancement of Artificial Intelligence (AI) techniques, the emerging Deep Learning (DL) models have been proving highly effective when forecasting this pandemic future from the huge historical data. However, the main weakness of DL models is lacking the explanation capabilities. To overcome this limitation, we introduce a novel combination of the Susceptible-Infectious-Recovered-Deceased (SIRD) compartmental model and Variational Autoencoder (VAE) neural network known as BeCaked. With pandemic data provided by the Johns Hopkins University Center for Systems Science and Engineering, our model achieves 0.98 \\R^2\\and 0.012 MAPE at world level with 31-step forecast and up to 0.99 \\R^2\\and 0.0026 MAPE at country level with 15-step forecast on predicting daily infectious cases. Not only enjoying high accuracy, but BeCaked also offers useful justifications for its results based on the parameters of the SIRD model. Therefore, BeCaked can be used as a reference for authorities or medical experts to make on time right decisions.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">nguyen_becaked_2022</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{{BeCaked}: {An} {Explainable} {Artificial} {Intelligence} {Model} for {COVID}-19 {Forecasting}}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{12}</span><span class="p">,</span>
  <span class="na">copyright</span> <span class="p">=</span> <span class="s">{Creative Commons Attribution-ShareAlike 4.0 International License}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{2045-2322}</span><span class="p">,</span>
  <span class="na">shorttitle</span> <span class="p">=</span> <span class="s">{{BeCaked}}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://www.nature.com/articles/s41598-022-11693-9}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1038/s41598-022-11693-9}</span><span class="p">,</span>
  <span class="na">language</span> <span class="p">=</span> <span class="s">{en}</span><span class="p">,</span>
  <span class="na">number</span> <span class="p">=</span> <span class="s">{1}</span><span class="p">,</span>
  <span class="na">urldate</span> <span class="p">=</span> <span class="s">{2025-08-27}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{Scientific Reports}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Nguyen, Duc Q. and Vo, Nghia Q. and Nguyen, Thinh T. and Nguyen-An, Khuong and Nguyen, Quang H. and Tran, Dang N. and Quan, Tho T.}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">may</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
  <span class="na">note</span> <span class="p">=</span> <span class="s">{Publisher: Nature Publishing Group}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Applied mathematics, Bioinformatics, Computational models, Computational science, Computer science, Differential equations, Disease model, Dynamical systems, Machine learning, Public health, Time series}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{7969}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> </div> <div id="nguyen_towards_2022" class="col-sm-8"> <div class="title">Towards De Novo Drug Design for the Coronavirus: A Drug-Target Interaction Prediction Approach using Atom-enhanced Graph Neural Network with Multi-hop Gating Mechanism</div> <div class="author"> <em>Duc Q. Nguyen</em>, Khoan D. Le, Bach T. Ly, and <span class="more-authors" title="click to view 7 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '7 more authors' ? 'An D. Nguyen, Quang H. Nguyen, Tuan H. Nguyen, Tho T. Quan, Cuong Quoc Duong, Phuong Thuy Viet Nguyen, Thanh N. Truong' : '7 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">7 more authors</span> </div> <div class="periodical"> <em>In 2022 9th NAFOSTED Conference on Information and Computer Science (NICS)</em>, Oct 2022 </div> <div class="periodical"> </div> <div class="links"> <a class="award btn btn-sm z-depth-0" role="button">Best Paper Award</a> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1109/NICS56915.2022.10013437" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="/assets/pdf/papers/Nguyen%20et%20al.%20-%202022%20-%20Towards%20De%20Novo%20Drug%20Design%20for%20the%20Coronavirus%20A%20Drug-Target%20Interaction%20Prediction%20Approach%20using%20Atom-enhanced%20Graph%20Neural%20Network%20with%20Multi-hop%20Gating%20Mechanism.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="award hidden d-print-inline"> <p></p> <p>Best Paper Award in NICS 2022</p> </div> <div class="abstract hidden"> <p>For humans, the COVID-19 pandemic and Coronavirus have undeniably been a nightmare. Although there are effective vaccines, specific drugs are still urgent. Normally, to identify potential drugs, one needs to design and then test interactions between the drug and the virus in an in silico manner for determining candidates. This Drug-Target Interaction (DTI) process, can be done by molecular docking, which is too complicated and time-consuming for manual works. Therefore, it opens room for applying Artificial Intelligence (AI) techniques. In particular, Graph Neural Network (GNN) attracts recent attention since its high suitability for the nature of drug compounds and virus proteins. However, to introduce such a representation well-reflecting biological structures of biological compounds is not a trivial task. Moreover, since available datasets of Coronavirus are still not highly popular, the recently developed GNNs have been suffering from overfitting on them. We then address those issues by proposing a novel model known as Atom-enhanced Graph Neural Network with Multi-hop Gating Mechanism. On one hand, our model can learn more precise features of compounds and proteins. On the other hand, we introduce a new gating mechanism to create better atom representation from non-neighbor information. Once applying transfer learning from very large databanks, our model enjoys promising performance, especially when experimenting with Coronavirus.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">nguyen_towards_2022</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Towards {De} {Novo} {Drug} {Design} for the {Coronavirus}: {A} {Drug}-{Target} {Interaction} {Prediction} {Approach} using {Atom}-enhanced {Graph} {Neural} {Network} with {Multi}-hop {Gating} {Mechanism}}</span><span class="p">,</span>
  <span class="na">copyright</span> <span class="p">=</span> <span class="s">{Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License}</span><span class="p">,</span>
  <span class="na">shorttitle</span> <span class="p">=</span> <span class="s">{Towards {De} {Novo} {Drug} {Design} for the {Coronavirus}}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://ieeexplore.ieee.org/document/10013437}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/NICS56915.2022.10013437}</span><span class="p">,</span>
  <span class="na">urldate</span> <span class="p">=</span> <span class="s">{2025-09-08}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{2022 9th {NAFOSTED} {Conference} on {Information} and {Computer} {Science} ({NICS})}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Nguyen, Duc Q. and Le, Khoan D. and Ly, Bach T. and Nguyen, An D. and Nguyen, Quang H. and Nguyen, Tuan H. and Quan, Tho T. and Duong, Cuong Quoc and Nguyen, Phuong Thuy Viet and Truong, Thanh N.}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">oct</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2022}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{Coronavirus, COVID-19, Drug-Target Interaction, Drugs, Graph neural networks, Graph Neural Networks, Multi-hop Gating Mechanism, Proteins, Spread spectrum communication, Transfer learning, Vaccines}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{275--280}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100" style="background-color:orangered"> <a href="https://naacl.org/" rel="external nofollow noopener" target="_blank">NAACL</a> </abbr> </div> <div id="truong_crossing_2024" class="col-sm-8"> <div class="title">Crossing Linguistic Horizons: Finetuning and Comprehensive Evaluation of Vietnamese Large Language Models</div> <div class="author"> <a href="https://cs.stanford.edu/~sttruong/" rel="external nofollow noopener" target="_blank">Sang Truong</a>, <em>Duc Nguyen</em>, Toan Nguyen, and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Dong Le, Nhi Truong, Tho Quan, Sanmi Koyejo' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>In Findings of the Association for Computational Linguistics: NAACL 2024</em>, Jun 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.18653/v1/2024.findings-naacl.182" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="/assets/pdf/papers/Truong%20et%20al.%20-%202024%20-%20Crossing%20Linguistic%20Horizons%20Finetuning%20and%20Comprehensive%20Evaluation%20of%20Vietnamese%20Large%20Language%20Models.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p>Recent advancements in large language models (LLMs) have underscored their importance in the evolution of artificial intelligence. However, despite extensive pretraining on multilingual datasets, available open-sourced LLMs exhibit limited effectiveness in processing Vietnamese. The challenge is exacerbated by the absence of systematic benchmark datasets and metrics tailored for Vietnamese LLM evaluation. To mitigate these issues, we have finetuned LLMs specifically for Vietnamese and developed a comprehensive evaluation framework encompassing 10 tasks and 31 metrics. We observe that finetuning can help LLMs transfer knowledge across languages, serving as an efficient way to bolster their capabilities in non-English languages. Moreover, our analysis indicates that larger models can introduce more biases and uncalibrated outputs and the key factor influencing LLM performance is the quality of the training or finetuning datasets. These insights underscore the significance of meticulous finetuning with high-quality datasets in enhancing LLM performance.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">truong_crossing_2024</span><span class="p">,</span>
  <span class="na">address</span> <span class="p">=</span> <span class="s">{Mexico City, Mexico}</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Crossing {Linguistic} {Horizons}: {Finetuning} and {Comprehensive} {Evaluation} of {Vietnamese} {Large} {Language} {Models}}</span><span class="p">,</span>
  <span class="na">copyright</span> <span class="p">=</span> <span class="s">{Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License}</span><span class="p">,</span>
  <span class="na">shorttitle</span> <span class="p">=</span> <span class="s">{Crossing {Linguistic} {Horizons}}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://aclanthology.org/2024.findings-naacl.182/}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.18653/v1/2024.findings-naacl.182}</span><span class="p">,</span>
  <span class="na">urldate</span> <span class="p">=</span> <span class="s">{2025-09-08}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Findings of the {Association} for {Computational} {Linguistics}: {NAACL} 2024}</span><span class="p">,</span>
  <span class="na">publisher</span> <span class="p">=</span> <span class="s">{Association for Computational Linguistics}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Truong, Sang and Nguyen, Duc and Nguyen, Toan and Le, Dong and Truong, Nhi and Quan, Tho and Koyejo, Sanmi}</span><span class="p">,</span>
  <span class="na">editor</span> <span class="p">=</span> <span class="s">{Duh, Kevin and Gomez, Helena and Bethard, Steven}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">jun</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{2849--2900}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100" style="background-color:navy"> <a href="https://ieeeaccess.ieee.org/" rel="external nofollow noopener" target="_blank">IEEE Access</a> </abbr> </div> <div id="nguyen_explainable_2024" class="col-sm-8"> <div class="title">Explainable Neural Subgraph Matching With Learnable Multi-Hop Attention</div> <div class="author"> <em>Duc Q. Nguyen</em>, Thanh Toan Nguyen, Jun Jo, and <span class="more-authors" title="click to view 3 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '3 more authors' ? 'Florent Poux, Shikha Anirban, Tho T. Quan' : '3 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">3 more authors</span> </div> <div class="periodical"> <em>IEEE Access</em>, Jun 2024 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a href="https://doi.org/10.1109/ACCESS.2024.3458050" class="btn btn-sm z-depth-0" role="button" rel="external nofollow noopener" target="_blank">DOI</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="/assets/pdf/papers/Nguyen%20et%20al.%20-%202024%20-%20Explainable%20Neural%20Subgraph%20Matching%20With%20Learnable%20Multi-Hop%20Attention.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p>Subgraph matching is a challenging problem with a wide range of applications in drug discovery, social network analysis, biochemistry, and cognitive science. It involves determining whether a given query graph is present within a larger target graph. Traditional graph-matching algorithms provide precise results but face challenges in large graph instances due to the NP-complete nature of the problem, limiting their practical applicability. In contrast, recent neural network-based approximations offer more scalable solutions but often lack interpretable node correspondences. To address these limitations, this article presents a multi-task learning framework called xNeuSM: Explainable Neural Subgraph Matching, which introduces Graph Learnable Multi-hop Attention Networks (GLeMA) that adaptively learn the parameters governing the attention factor decay for each node across hops rather than relying on fixed hyperparameters. Our framework jointly optimizes both subgraph matching and finding subgraph-isomorphism mappings. We provide a theoretical analysis establishing error bounds for GLeMA‚Äôs approximation of multi-hop attention as a function of the number of hops. Additionally, we prove that learning distinct attention decay factors for each node leads to a correct approximation of multi-hop attention. Empirical evaluation on real-world datasets shows that xNeuSM achieves substantial improvements in prediction F1 score of up to 34% compared to approximate baselines and, notably, at least a seven-fold faster query time than exact algorithms. With these results, xNeuSM can be applied to solve matching problems in various domains spanning from biochemistry to social science.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@article</span><span class="p">{</span><span class="nl">nguyen_explainable_2024</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Explainable {Neural} {Subgraph} {Matching} {With} {Learnable} {Multi}-{Hop} {Attention}}</span><span class="p">,</span>
  <span class="na">volume</span> <span class="p">=</span> <span class="s">{12}</span><span class="p">,</span>
  <span class="na">copyright</span> <span class="p">=</span> <span class="s">{Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License}</span><span class="p">,</span>
  <span class="na">issn</span> <span class="p">=</span> <span class="s">{2169-3536}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://ieeexplore.ieee.org/document/10677500}</span><span class="p">,</span>
  <span class="na">doi</span> <span class="p">=</span> <span class="s">{10.1109/ACCESS.2024.3458050}</span><span class="p">,</span>
  <span class="na">urldate</span> <span class="p">=</span> <span class="s">{2025-09-08}</span><span class="p">,</span>
  <span class="na">journal</span> <span class="p">=</span> <span class="s">{IEEE Access}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Nguyen, Duc Q. and Toan Nguyen, Thanh and Jo, Jun and Poux, Florent and Anirban, Shikha and Quan, Tho T.}</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2024}</span><span class="p">,</span>
  <span class="na">keywords</span> <span class="p">=</span> <span class="s">{graph neural networks, Graph neural networks, Spread spectrum communication, Pattern matching, subgraph matching, Adaptation models, Explainability, learnable multi-hop attention, Multitasking, Object recognition}</span><span class="p">,</span>
  <span class="na">pages</span> <span class="p">=</span> <span class="s">{130474--130492}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> <li> <div class="row"> <div class="col col-sm-2 abbr"> <abbr class="badge rounded w-100" style="background-color:teal"> <a href="https://iclr.cc/" rel="external nofollow noopener" target="_blank">ICLR</a> </abbr> </div> <div id="truong_neural_2025" class="col-sm-8"> <div class="title">Neural Nonmyopic Bayesian Optimization in Dynamic Cost Settings</div> <div class="author"> <a href="https://cs.stanford.edu/~sttruong/" rel="external nofollow noopener" target="_blank">Sang T. Truong</a>, <em>Duc Q. Nguyen</em>, Willie Neiswanger, and <span class="more-authors" title="click to view 4 more authors" onclick=" var element=$(this); element.attr('title', ''); var more_authors_text=element.text() == '4 more authors' ? 'Ryan-Rhys Griffiths, Stefano Ermon, Nick Haber, Sanmi Koyejo' : '4 more authors'; var cursorPosition=0; var textAdder=setInterval(function(){ element.html(more_authors_text.substring(0, cursorPosition + 1)); if (++cursorPosition == more_authors_text.length){ clearInterval(textAdder); } }, '10'); ">4 more authors</span> </div> <div class="periodical"> <em>In Towards Agentic AI for Science: Hypothesis Generation, Comprehension, Quantification, and Validation</em>, Mar 2025 </div> <div class="periodical"> </div> <div class="links"> <a class="abstract btn btn-sm z-depth-0" role="button">Abs</a> <a class="bibtex btn btn-sm z-depth-0" role="button">Bib</a> <a href="/assets/pdf/papers/Truong%20et%20al.%20-%202025%20-%20Neural%20Nonmyopic%20Bayesian%20Optimization%20in%20Dynamic%20Cost%20Settings.pdf" class="btn btn-sm z-depth-0" role="button">PDF</a> </div> <div class="abstract hidden"> <p>Bayesian optimization (BO) is a popular framework for optimizing black-box functions, leveraging probabilistic models such as Gaussian processes. Conventional BO algorithms, however, assume static query costs, which limit their applicability to real-world problems with dynamic cost structures such as geological surveys or biological sequence design, where query costs vary based on the previous actions. We propose a novel nonmyopic BO algorithm named LookaHES featuring dynamic cost models to address this. LookaHES employs a neural network policy for variational optimization over multi-step lookahead horizons to enable planning under dynamic cost environments. Empirically, we benchmark LookaHES on synthetic functions exhibiting varied dynamic cost structures. We subsequently apply LookaHES to a real-world application in protein sequence design using a large language model policy, demonstrating its scalability and effectiveness in handling multi-step planning in a large and complex query space. LookaHES consistently outperforms its myopic counterparts in synthetic and real-world settings, significantly improving efficiency and solution quality. Our implementation is available at https://github.com/sangttruong/nonmyopia.</p> </div> <div class="bibtex hidden"> <figure class="highlight"><pre><code class="language-bibtex" data-lang="bibtex"><span class="nc">@inproceedings</span><span class="p">{</span><span class="nl">truong_neural_2025</span><span class="p">,</span>
  <span class="na">title</span> <span class="p">=</span> <span class="s">{Neural {Nonmyopic} {Bayesian} {Optimization} in {Dynamic} {Cost} {Settings}}</span><span class="p">,</span>
  <span class="na">copyright</span> <span class="p">=</span> <span class="s">{Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License}</span><span class="p">,</span>
  <span class="na">url</span> <span class="p">=</span> <span class="s">{https://openreview.net/forum?id=DFCT9dHkko}</span><span class="p">,</span>
  <span class="na">language</span> <span class="p">=</span> <span class="s">{en}</span><span class="p">,</span>
  <span class="na">urldate</span> <span class="p">=</span> <span class="s">{2025-09-08}</span><span class="p">,</span>
  <span class="na">booktitle</span> <span class="p">=</span> <span class="s">{Towards {Agentic} {AI} for {Science}: {Hypothesis} {Generation}, {Comprehension}, {Quantification}, and {Validation}}</span><span class="p">,</span>
  <span class="na">author</span> <span class="p">=</span> <span class="s">{Truong, Sang T. and Nguyen, Duc Q. and Neiswanger, Willie and Griffiths, Ryan-Rhys and Ermon, Stefano and Haber, Nick and Koyejo, Sanmi}</span><span class="p">,</span>
  <span class="na">month</span> <span class="p">=</span> <span class="nv">mar</span><span class="p">,</span>
  <span class="na">year</span> <span class="p">=</span> <span class="s">{2025}</span><span class="p">,</span>
<span class="p">}</span></code></pre></figure> </div> </div> </div> </li> </ol> </div> <div class="social"> <div class="contact-icons"> <a href="/assets/pdf/cv-nqduc.pdf" title="CV" target="_blank"><i class="ai ai-cv"></i></a> <a href="mailto:%6E%71%64%75%63@%75.%6E%75%73.%65%64%75" title="email"><i class="fa-solid fa-envelope"></i></a> <a href="https://github.com/martinakaduc" title="GitHub" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-github"></i></a> <a href="https://www.linkedin.com/in/martinakaduc" title="LinkedIn" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-linkedin"></i></a> <a href="https://orcid.org/0000-0001-9638-8289" title="ORCID" rel="external nofollow noopener" target="_blank"><i class="ai ai-orcid"></i></a> <a href="https://www.researchgate.net/profile/Quang-Duc-Nguyen/" title="ResearchGate" rel="external nofollow noopener" target="_blank"><i class="ai ai-researchgate"></i></a> <a href="https://scholar.google.com/citations?user=w1o63XAAAAAJ" title="Google Scholar" rel="external nofollow noopener" target="_blank"><i class="ai ai-google-scholar"></i></a> <a href="https://twitter.com/martinakaduc" title="X" rel="external nofollow noopener" target="_blank"><i class="fa-brands fa-x-twitter"></i></a> </div> <div class="contact-note">Feel free to reach out via email for any collaboration or research inquiries. </div> </div> </article> </div> </div> <footer class="fixed-bottom" role="contentinfo"> <div class="container mt-0"> ¬© Copyright 2026 Duc Q. Nguyen. Powered by <a href="https://jekyllrb.com/" target="_blank" rel="external nofollow noopener">Jekyll</a> with <a href="https://github.com/alshedivat/al-folio" rel="external nofollow noopener" target="_blank">al-folio</a> theme. Hosted by <a href="https://pages.github.com/" target="_blank" rel="external nofollow noopener">GitHub Pages</a>. </div> </footer> <script src="https://cdn.jsdelivr.net/npm/jquery@3.6.0/dist/jquery.min.js" integrity="sha256-/xUj+3OJU5yExlq6GSYGSHk7tPXikynS7ogEvDej/m4=" crossorigin="anonymous"></script> <script src="/assets/js/bootstrap.bundle.min.js"></script> <script src="https://cdn.jsdelivr.net/npm/mdbootstrap@4.20.0/js/mdb.min.js" integrity="sha256-NdbiivsvWt7VYCt6hYNT3h/th9vSTL4EDWeGs5SN3DA=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/masonry-layout@4.2.2/dist/masonry.pkgd.min.js" integrity="sha256-Nn1q/fx0H7SNLZMQ5Hw5JLaTRZp0yILA/FRexe19VdI=" crossorigin="anonymous"></script> <script defer src="https://cdn.jsdelivr.net/npm/imagesloaded@5.0.0/imagesloaded.pkgd.min.js" integrity="sha256-htrLFfZJ6v5udOG+3kNLINIKh2gvoKqwEhHYfTTMICc=" crossorigin="anonymous"></script> <script defer src="/assets/js/masonry.js?a0db7e5d5c70cc3252b3138b0c91dcaf" type="text/javascript"></script> <script defer src="https://cdn.jsdelivr.net/npm/medium-zoom@1.1.0/dist/medium-zoom.min.js" integrity="sha256-ZgMyDAIYDYGxbcpJcfUnYwNevG/xi9OHKaR/8GK+jWc=" crossorigin="anonymous"></script> <script defer src="/assets/js/zoom.js?85ddb88934d28b74e78031fd54cf8308"></script> <script src="/assets/js/no_defer.js?2781658a0a2b13ed609542042a859126"></script> <script defer src="/assets/js/common.js?c15de51d4bb57887caa2c21988d97279"></script> <script defer src="/assets/js/copy_code.js?c8a01c11a92744d44b093fc3bda915df" type="text/javascript"></script> <script defer src="/assets/js/jupyter_new_tab.js?d9f17b6adc2311cbabd747f4538bb15f"></script> <script async src="https://d1bxh8uas1mnw7.cloudfront.net/assets/embed.js"></script> <script async src="https://badge.dimensions.ai/badge.js"></script> <script defer type="text/javascript" id="MathJax-script" src="https://cdn.jsdelivr.net/npm/mathjax@3.2.2/es5/tex-mml-chtml.js" integrity="sha256-MASABpB4tYktI2Oitl4t+78w/lyA+D7b/s9GEP0JOGI=" crossorigin="anonymous"></script> <script src="/assets/js/mathjax-setup.js?a5bb4e6a542c546dd929b24b8b236dfd"></script> <script defer src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6" crossorigin="anonymous"></script> <script defer src="/assets/js/progress-bar.js?2f30e0e6801ea8f5036fa66e1ab0a71a" type="text/javascript"></script> <script src="/assets/js/vanilla-back-to-top.min.js?f40d453793ff4f64e238e420181a1d17"></script> <script>
    addBackToTop();
  </script> <script type="module" src="/assets/js/search/ninja-keys.min.js?a3446f084dcaecc5f75aa1757d087dcf"></script> <ninja-keys hidebreadcrumbs noautoloadmdicons placeholder="Type to start searching"></ninja-keys> <script src="/assets/js/search-setup.js?6c304f7b1992d4b60f7a07956e52f04a"></script> <script src="/assets/js/search-data.js"></script> <script src="/assets/js/shortcut-key.js?6f508d74becd347268a7f822bca7309d"></script> </body> </html>